spec/ddtrace/transport_spec.rb

```diff
@@ -0,0 +1,127 @@
+require 'spec_helper'
+
+require 'ddtrace'
+
+RSpec.describe Datadog::HTTPTransport do
+  let(:transport) { described_class.new('localhost', '8126', options) }
+  let(:options) { {} }
+
+  before(:each) do
+    @original_level = Datadog::Tracer.log.level
+    Datadog::Tracer.log.level = Logger::FATAL
+  end
+
+  after(:each) do
+    Datadog::Tracer.log.level = @original_level
+  end
+
+  describe '#handle_response' do
+    subject(:result) { transport.handle_response(response) }
+
+    context 'given an OK response' do
+      let(:response) { Net::HTTPResponse.new(1.0, 200, 'OK') }
```

Comment:
just a a thought: sometimes adding some randomness to the test might make it more robust without requiring a exhaustive list of test cases. in case of this method you could do something like:
```rb
let(:status_code) { (200...500).to_a.sample }
let(:response) { Net::HTTPResponse.new(1.0, status_code, 'whatever') }
```
and test if the return value of the method is equal to `status_code`

Reply:
I think that works if the expected behavior is abstracted away from the input (e.g. it always replies with whatever input I gave it.)